### ISSUE TITLE
no-extra-parens does not warn around spread arguments

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#8175]
**Tell us about your environment**

* **ESLint Version:** 3.16.1
* **Node Version:** 7.6.0
* **npm Version:** 4.2.0

**What parser (default, Babel-ESLint, etc.) are you using?**

- default

**Please show your full configuration:**

- nothing

**What did you do? Please include the actual source code causing the issue.**

I checked the following code:

```js
/*eslint no-extra-parens: [error, all] */

let a = [...(b)]
let c = {...(d)}
```

```bash
$ eslint test.js --parser-options "ecmaVersion:2015,ecmaFeatures:{experimentalObjectRestSpread:true}" --no-eslintrc --no-ignore
```

**What did you expect to happen?**

ESLint warns the parentheses.

**What actually happened? Please include the actual, raw output from ESLint.**

No errors.
